tests/em/static/test_DC_FieldsDipoleFullspace.py

```diff
@@ -0,0 +1,308 @@
+from __future__ import print_function
+import unittest
+from SimPEG import Mesh, Utils, EM
+import numpy as np
+import SimPEG.EM.Static.DC as DC
+try:
+    from pymatsolver import Pardiso as Solver
+except ImportError:
+    from SimPEG import SolverLU as Solver
+
+from geoana import utils, spatial
+from geoana.em import fdem
+
+from geoana import utils, spatial
+from geoana.em import fdem
+from scipy.constants import mu_0, epsilon_0
+
+
+class DC_CC_DipoleFullspaceTests(unittest.TestCase):
+
+    def setUp(self):
+
+        cs = 0.5
+        npad = 21
+        hx = [(cs, npad, -1.2), (cs, 15), (cs, npad, 1.2)]
+        hy = [(cs, npad, -1.2), (cs, 15), (cs, npad, 1.2)]
+        hz = [(cs, npad, -1.2), (cs, 15), (cs, npad, 1.2)]
+        mesh = Mesh.TensorMesh([hx, hy, hz], x0="CCC")
+        sigma = np.ones(mesh.nC)*1e-2
+
+        # Set up survey parameters for numeric solution
+        x = mesh.vectorCCx[(mesh.vectorCCx > -75.) & (mesh.vectorCCx < 75.)]
+        y = mesh.vectorCCy[(mesh.vectorCCy > -75.) & (mesh.vectorCCy < 75.)]
+
+        Aloc = np.r_[1.0, 0., 0.]
+        Bloc = np.r_[-1.0, 0., 0.]
+        M = Utils.ndgrid(x-25., y, np.r_[0.])
+        N = Utils.ndgrid(x+25., y, np.r_[0.])
+
+        rx = DC.Rx.Dipole(M, N)
+        src = DC.Src.Dipole([rx], Aloc, Bloc)
+        survey = DC.Survey([src])
+
+        # Create Dipole Obj for Analytic Solution
+        edipole = fdem.ElectricDipoleWholeSpace(
```

Comment:
might need to add geoana to `requirements_dev.txt`. Nice to see this code in use :)